sdk/core/core-client/src/serviceClient.ts

```diff
@@ -211,9 +210,7 @@ function createDefaultPipeline(options: ServiceClientOptions): Pipeline {
 function getCredentialScopes(options: ServiceClientOptions): string | string[] | undefined {
   if (options.credentialScopes) {
     const scopes = options.credentialScopes;
-    return Array.isArray(scopes)
-      ? scopes.map((scope) => new URL(scope).toString())
-      : new URL(scopes).toString();
+    return Array.isArray(scopes) ? scopes.map((scope) => scope) : scopes;
```

Comment:
now we just need to return `scopes` as it maps to itself.

Reply:
Yeah you are right, no need to do all this :)